parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -0,0 +1,396 @@
+        """ Listens on the pending_result_queue and sends out results via 0mq
+
+        Parameters:
+        -----------
+        kill_event : threading.Event
+              Event to let the thread know when it is time to die.
+        """
+
+        # We set this timeout so that the thread checks the kill_event and does not
+        # block forever on the internal result queue
+        timeout = 0.1
+        # timer = time.time()
+        logger.debug("[RESULT_PUSH_THREAD] Starting thread")
+
+        while not kill_event.is_set():
+            time.sleep(LOOP_SLOWDOWN)
+            try:
+                result = self.pending_result_queue.get(block=True, timeout=timeout)
+                self.result_outgoing.send(result)
+                logger.debug("[RESULT_PUSH_THREAD] Sent result:{}".format(result))
+
+            except queue.Empty:
+                logger.debug("[RESULT_PUSH_THREAD] No results to send in past {}seconds".format(timeout))
+
+            except Exception as e:
+                logger.exception("[RESULT_PUSH_THREAD] Got an exception : {}".format(e))
+
+    def start(self):
+        """ Start the Manager process.
+
+        The worker loops on this:
+
+        1. If the last message sent was older than heartbeat period we send a heartbeat
+        2.
+
+
+        TODO: Move task receiving to a thread
+        """
+        self._kill_event = threading.Event()
+
+        self.procs = {}
+        for worker_id in range(self.worker_count):
+            p = multiprocessing.Process(target=worker, args=(worker_id,
+                                                             self.pending_task_queue,
+                                                             self.pending_result_queue,
+                                                             self.ready_worker_queue,
+                                                         ))
+            p.start()
+            self.procs[worker_id] = p
+        logger.debug("Manager synced with workers")
+
+        self._task_puller_thread = threading.Thread(target=self.pull_tasks,
+                                                    args=(self._kill_event,))
+        self._result_pusher_thread = threading.Thread(target=self.push_results,
+                                                      args=(self._kill_event,))
+        self._task_puller_thread.start()
+        self._result_pusher_thread.start()
+
+        start = None
+        abort_flag = False
+
+        logger.info("Loop start")
+
+        # TODO : Add mechanism in this loop to stop the fabric.
+        # This might need a multiprocessing event to signal back.
+        while not abort_flag:
+            time.sleep(0.1)
+
+        for proc_id in self.procs:
+            self.procs[proc_id].terminate()
+
+        # self._task_puller_thread.join()
+        # self._result_pusher_thread.join()
+        delta = time.time() - start
+        logger.info("Fabric ran for {} seconds".format(delta))
+        sys.exit(-1)
```

Comment:
is this an abnormal exit or the way this code is meant to exit? if this is how we're meant to exit, exit with 0, not -1.